Xamarin.PropertyEditing.Mac/PropertyTableDelegate.cs

```diff
@@ -140,9 +140,15 @@ public override void ItemDidExpand (NSNotification notification)
 				return;
 			}
 
-			if (facade.Target is PanelGroupViewModel group)
+			if (facade.Target is PanelGroupViewModel group) {
 				this.dataSource.DataContext.SetIsExpanded (group.Category, isExpanded: true);
-			else if (facade.Target is ObjectPropertyViewModel ovm) {
+
+				// Also update the expander state. This is needed when the property editor first
+				// displays and category groups default to expanded, to show the right icon
+				NSView view = outline.GetView (0, row, makeIfNecessary: false);
+				if (view?.Subviews != null && view.Subviews.Length > 0 && view.Subviews[0] is NSButton expander)
```

Comment:
Please don't query Subviews 3 times. It allocates an array on every pinvoke.
```suggestion
				if (view.Subviews is NSView[] subviews && subviews.Length > 0 && subviews is NSButton expander)
```

Reply:
Updated.